### DISCUSSION TITLE
how to prefill an input field in template using `Mojolicious::Plugin::TagHelpers`

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2185] [General]
I use `Mojolicious::Plugin::TagHelpers` to design my form templates, and I've fields like the following one:

```
        <div class="col">                                                                                 
            %= text_field 'name' => id => 'name', class => 'form-control'                                   
        </div>     
```

How can I prefill the field with the value of stashed variable `$user->name` if set? So I would like to inject a value like `$user ? $user->name : ''` but I cannot find how to instrument the `text_field` appropriately.

**Comment by daleif** [comment]
`text_field` takes the prefilled data from `$c->param`, so you use `$c->param( name => $user->name // '' );` in your controller.

**Comment by s1037989** [comment]
I think this is documented in Mojolicious::Plugin::TagHelpers
<https://docs.mojolicious.org/Mojolicious/Plugin/TagHelpers#toc>
 Description:

*Most form helpers can automatically pick up previous input values and will
show them as default. You can also use "param" in
Mojolicious::Plugin::DefaultHelpers to set them manually and let necessary
attributes always be generated automatically.*

The example includes this as well:

*% param country => 'germany' unless param 'country';* <%= radio_button country
=> 'germany' %> Germany <%= radio_button country => 'france' %> France <%=
radio_button country => 'uk' %> UK

**Comment by fluca1978** [comment]
@daleif thanks, sometimes I forgot how easy and consistent Mojolicious is!
@s1037989 thanks, I totally missed such paragraph in the documentation, shame on me.

**Reply by daleif** [reply]
It took me a while to figure that out as well at the beginning.

Do note though that there is a caveat: There are some terms in `param` that are used for other things, so some names cannot be used for form variable names.

You seem to be using Bootstrap or similar (I do as well). I've been thinking whether it might be useful to make a version of `Mojolicious::Plugin::TagHelpers` where certain tags already come with the normal BS classes. And perhaps have two class attributes, what that adds to the default class setting, and one that overwrites (like with `content_for` and `content_with`)

**Reply by fluca1978** [reply]
> Do note though that there is a caveat: There are some terms in `param` that are used for other things, so some names cannot be used for form variable names.

Yes, clear.

> 
> You seem to be using Bootstrap or similar (I do as well). I've been thinking whether it might be useful to make a version of `Mojolicious::Plugin::TagHelpers` where certain tags already come with the normal BS classes. And perhaps have two class attributes, what that adds to the default class setting, and one that overwrites (like with `content_for` and `content_with`)

I'm using bootstrap, and it would be great to have such an integration in Mojolicious, according to me.
